camus-kafka-coders/src/main/java/com/linkedin/camus/etl/kafka/coders/AvroMessageDecoder.java

```diff
@@ -0,0 +1,120 @@
+package com.linkedin.camus.etl.kafka.coders;
+
+import com.linkedin.camus.coders.CamusWrapper;
+import com.linkedin.camus.coders.MessageDecoder;
+import com.linkedin.camus.coders.MessageDecoderException;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericData.Record;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.util.Utf8;
+import org.apache.hadoop.io.Text;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Properties;
+
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+
+public class AvroMessageDecoder extends MessageDecoder<byte[], GenericData.Record> {
+  private static final byte MAGIC_BYTE = 0x0;
+  private static final int idSize = 4;
+  private final String SCHEMA_REGISTRY_URL = "schema.registry.url";
+  private final String MAX_SCHEMAS_PER_SUBJECT = "max.schemas.per.subject";
+  private final String DEFAULT_MAX_SCHEMAS_PER_SUBJECT = "1000";
+  private static final Logger logger = Logger.getLogger(AvroMessageDecoder.class);
+  protected DecoderFactory decoderFactory;
+  private SchemaRegistryClient schemaRegistry;
+
+  @Override
+  public void init(Properties props, String topicName) {
+    super.init(props, topicName);
+    decoderFactory = DecoderFactory.get();
+    if (props == null) {
+      throw new IllegalArgumentException("Missing schema registry url!");
+    }
+    String url = props.getProperty(SCHEMA_REGISTRY_URL);
+    if (url == null) {
+      throw new IllegalArgumentException("Missing schema registry url!");
+    }
+    String maxSchemaObject = props.getProperty(
+        MAX_SCHEMAS_PER_SUBJECT, DEFAULT_MAX_SCHEMAS_PER_SUBJECT);
+    schemaRegistry = new CachedSchemaRegistryClient(url, Integer.parseInt(maxSchemaObject));
+  }
+
+  private ByteBuffer getByteBuffer(byte[] payload) {
+    ByteBuffer buffer = ByteBuffer.wrap(payload);
+    byte magic = buffer.get();
+    logger.info("MAGIC BYTE" + magic);
+    if (magic != MAGIC_BYTE) {
+      throw new MessageDecoderException("Unknown magic byte!");
+    }
+    return buffer;
+  }
+
+  private Object deserialize(byte[] payload) throws MessageDecoderException {
+    try {
+      ByteBuffer buffer = getByteBuffer(payload);
+      int id = buffer.getInt();
+      Schema schema = schemaRegistry.getByID(id);
+      logger.info(schema.toString());
+      if (schema == null)
```

Comment:
The null check should be done before logging. Otherwise, schema.toString will hit NullPointerException.